internal/c/parts/audio/audio.cpp

```diff
@@ -1523,18 +1523,39 @@ void sub__sndplay(int32_t handle) {
 /// </summary>
 /// <param name="handle">A sound handle to copy</param>
 /// <param name="volume">The volume at which the sound should be played (0.0 - 1.0)</param>
+/// <param name="x">x distance values go from left (negative) to right (positive)</param>
+/// <param name="y">y distance values go from below (negative) to above (positive).</param>
+/// <param name="z">z distance values go from behind (negative) to in front (positive).</param>
 /// <param name="passed">How many parameters were passed?</param>
-void sub__sndplaycopy(int32_t src_handle, double volume, int32_t passed) {
+void sub__sndplaycopy(int32_t src_handle, double volume, double x, double y, double z, int32_t passed) {
     // We are simply going to use sndcopy, then setup some stuff like volume and autokill and then use sndplay
     // We are not checking if the audio engine was initialized because if not we'll get an invalid handle anyway
     int32_t dst_handle = func__sndcopy(src_handle);
 
     // Check if we succeeded and then proceed
     if (dst_handle > 0) {
         // Set the volume if requested
-        if (passed)
+        if (passed & 1)
             ma_sound_set_volume(&audioEngine.soundHandles[dst_handle]->maSound, volume);
 
+        if (passed & 4 || passed & 8) {                                                                   // If y or z or both are passed
```

Comment:
Is it an option to just call `sub__sndbal()` here? Since we're basically just duplicating the logic.

Reply:
I was actually tempted to just call `sub__sndbal` and be done with it. However, I realized that we were going to set balance just once for the sound copy and the sound would eventually get destroyed. So, we do not need to care about what the old `x, y, z` was like `sub__sndbal` does and these are just set to zero if they are not passed. Also, `x` is completely optional unlike `sub__sndbal`.